go/libraries/doltcore/table/editor/pk_table_editor.go

```diff
@@ -476,6 +477,53 @@ func (te *pkTableEditor) InsertRow(ctx context.Context, dRow row.Row, errFunc PK
 	return te.InsertKeyVal(ctx, key.(types.Tuple), val.(types.Tuple), tagToVal, errFunc)
 }
 
+func (te *pkTableEditor) DeleteByKey(ctx context.Context, key types.Tuple, tagToVal map[uint64]types.Value) (retErr error) {
```

Comment:
Since we have this, I think we should do similar to `InsertRow` and `InsertKeyVal` and have `DeleteRow` just call this so that we're not duplicating a lot of code.